### PR TITLE
version 0.2.2 with snippets, indentation rules, code folding, updated README

### LINKED ISSUE #1: Add proper indentation settings
At the moment extension relies on the built-in "guess indentation" functionality. Unfortunately that doesn't work well with files created "from scratch".

### LINKED ISSUE #4: Remove deprecated `<-` assign operator in the `highlight-example.png`
The `<-` assignment operator was deprecated long time ago in Stan and it warns most users when they compile the Stan Code/Model.

I think the picture in README.md must be updated to only assign stuff with `=` assignment operator.

### PR DESCRIPTION
**Pull request by storopoli**
- Snippets
  - Control Flow: `if`, `else`, `elseif`, `for`, `ifelse`
  - Profile: `profile`
  - Program Blocks: `data`, `transformed data`, `parameters`, `transformed parameters`, `model`, `generated quantities`, `functions`
  - include Directive: `include`
- Added indentation rules (fixes #1 )
- Added better support to code folding
- Better support for the `#include` directive (better highlighting for files inside `""`)
- Removed old tmLanguage XML files and adopted tmLanguage.json new formats
- Added `invalid.illegal.variable` for variables named after Program Blocks (`data`, `model` etc.) (fixes the Known Issues header in README)
- Improved README with GIFs showing features and an updated syntax highlighting example (fixes #4)
- Organized stuff in folders

### CODE REVIEW

**Comment by storopoli** [comment]
@ivan-bocharov I've made a lot of changes and tested all. Please see the new GIFs for the new features.

File: package.json [hunk 1]
```diff
@@ -2,15 +2,15 @@
...
+    "version": "0.2.2",
     "publisher": "ivan-bocharov",
     "repository": "https://github.com/bocharov-ivan/stan-vscode",
     "icon": "img/stan_logo.png",
     "engines": {
         "vscode": "^1.5.0"
     },
     "categories": [
-        "Languages"
+        "Programming Languages", "Snippets"
```
Comment (storopoli) [inline package.json:13]: This is the new format from vscode. Instead of "Languages", we use "Programming Languages".

**Comment by storopoli** [comment]
@ivan-bocharov have you managed to take a look at this pull request?

**Comment by ivan-bocharov** [comment]
Hi @storopoli 

It was a busy week and I didn't have a chance to look at it before. This PR looks absolutely amazing! Thanks a lot for the effort.
I will probably re-record the videos with functionality demonstration because in my opinion these are a bit too fast and end a bit abruptly.

If you want to use your fixes, you can build the extension locally using `vsce` tool. I will try to publish a new version later this week.

**Comment by rok-cesnovar** [comment]
Great stuff @storopoli and @ivan-bocharov obviously as well. 

One minor request. Could we also add `array` as a keyword as well?

`array[N] real a;` is equivalent to the existing `real a[N]`
as well as
`array[N] vector[M] a;` to `vector[M] a[N];`

**Comment by ivan-bocharov** [comment]
Hi @rok-cesnovar, thanks for your kind words.

I've added `array` to the `variable types` block in grammar file.

**Comment by ivan-bocharov** [comment]
Hey @storopoli, I have added a capture for the block keywords (it was giving wrong highlighting on some of the themes) and re-recorded the demo videos using a slightly more neutral theme. What do you think? If you like these, I can push the changes to the PR and publish a new version, otherwise we can iterate on these.

https://user-images.githubusercontent.com/706421/108736025-7ca57e80-7531-11eb-9d71-6dea0294c574.mov


https://user-images.githubusercontent.com/706421/108736053-83cc8c80-7531-11eb-82aa-1590d9eaf3e4.mov

syntax

**Comment by storopoli** [comment]
It look awesome! Way better than mine (I'm not a good graphics guy). Sure, we can use that.

**Comment by ivan-bocharov** [comment]
Thanks again for all the contributions! I have published the new version and it should be available through the marketplace now.

**Comment by storopoli** [comment]
Great!!!

**Pull request merged**